src/persist/src/indexed/mod.rs

```diff
@@ -161,7 +169,48 @@ impl<K: Data, V: Data, U: Buffer, L: Blob> Indexed<K, V, U, L> {
         self.traces
             .entry(id)
             .or_insert_with_key(|id| BlobTrace::new(BlobTraceMeta::new(*id)));
-        id
+        Ok(id)
+    }
+
+    /// Removes a stream from the index.
+    ///
+    /// This method is idempotent and may be called multiple times. It returns
+    /// true if the stream was actually destroyed, and false if it was already
+    /// destroyed previously.
+    pub fn destroy(&mut self, id_str: &str) -> Result<bool, Error> {
+        if self
+            .graveyard
+            .iter()
+            .find(|(destroyed_name, _)| destroyed_name == &id_str)
```

Comment:
nit: `.find(...).is_some()` -> `.any(...)`

Reply:
fixed here and in the other place